Doc for second electrodialysis flowsheet (#1386)
watertap-org/watertap: docs/technical_reference/flowsheets/electrodialysis_1stack.rst

    * `One-stack electrodialysis with concentrate stream recirculation (feed-and-bleed mode) <https://watertap--1385.org.readthedocs.build/en/1385/technical_reference/flowsheets/electrodialysis_1stack_conc_recirc.html>`_.

Implementation
--------------

The modeled one-stack ED system is illustrated by Figure 1. The feed solution is split into two fluids through a separator unit, entering the diluate and concentrate channels of the ED stack. On the outlet side of the ED stack, all diluate fluids are collected into the total product water, and the concentrate fluids into the total brine stream. Water recovery in this model is volume-based, i.e., the ratio of product volume to total volume of feed solution. The model simulates the steady state of the ED system. The flowsheet relies on the following key assumptions:

   * supports steady-state only

>    * a property pacakge (i.e., MCAS) is provided for all unit models

Reviewer: ```suggestion
   * a property package (i.e., MCAS) is provided for all unit models
```

.. figure:: ../../_static/flowsheets/edfs.png
    :width: 500
    :align: center

    Figure 1. Flowsheet diagram: one-stack ED system

The electrodialysis 1D block is set up with the following configuration arguments:

.. code-block::

   m.fs.EDstack = Electrodialysis1D(
        property_package=m.fs.properties,
        operation_mode="Constant_Voltage",
        finite_elements=20,
    )

> These configurations enable the electrodialysis unit to use a flowsheet-unified property package, set a constant stack voltage, and adopt a favorable number of finite elements for 1-dimensional simulation and solving.. 

Reviewer 2: ```suggestion
These configurations enable the electrodialysis unit to use a flowsheet-unified property package, set a constant stack voltage, and adopt a favorable number of finite elements for 1-dimensional simulation and solving.
```

    * stack voltage applied
    * ED cell pair number 

In the given optimization case, the objective function is to minimize the levelized cost of water, which can be represented by the following equation
where :math:`Q` represents volumetric flow, :math:`f_{crf}` represents capital recovery factor
:math:`C_{cap,tot}` represents total capital cost, :math:`C_{op,tot}` represents total operating cost, and
:math:`f_{util}` represents the utilization factor:

    .. math::

        LCOW_{Q} = \frac{f_{crf}   C_{cap,tot} + C_{op,tot}}{f_{util} Q}

> The product water salinity is set to 1 :math:`g L^{-1}` (from a feed salinity of 9.9 :math:`g L^{-1}`).

Reviewer 2: I think this should be moved from the DOF section to the end of the Implementation section

> One-stack electrodialysis 

Reviewer 2: ```suggestion
One-Stack Electrodialysis 
```
Just to be consistent with the capitalization of the other flowsheets

=========================

Introduction
------------

> Electrodialysis (ED) is a promising technology for desalinating brackish waters and has been deployed at industrial scales [1]_. It utilizes electrical potential to drive ion diffusion through anion and cation exchange membranes, resulting in the dilution of the feed stream while producing a concentrated brine. A single ED stack is an assembly of multiple flow-by cells separated by alternating cation and anion exchange membranes positioned between a pair of electrodes. When a voltage is applied, ions in the cell pair are driven from one channel (forming the diluate channel) to the other (forming the concentrate channel), thereby desalinating water. A one-stack ED desalination system represents a basic ED operation, from which more complicated or larger-scale systems can be derived. Analyzing a one-stack ED system therefore provides valuable information on the technology's performance and cost-effectiveness for a given treatment task.

Reviewer 2: ```suggestion
Electrodialysis (ED) is a promising technology for desalinating brackish waters and has been deployed at industrial scales [1]_. It utilizes electrical potential to drive ion diffusion through anion and cation exchange membranes, resulting in the dilution of the feed stream while producing a concentrated brine. A single ED stack is an assembly of multiple flow-by cells separated by alternating cation and anion exchange membranes positioned between a pair of electrodes. When a voltage is applied, ions in the cell pair are driven from one channel (forming the diluate channel) to the other (forming the concentrate channel), thereby desalinating water. A one-stack ED desalination system represents a basic ED operation from which more complicated or larger-scale systems can be derived. Analyzing a one-stack ED system therefore provides valuable information on the technology's performance and cost-effectiveness for a given treatment task.
```

Equal flow conditions through the diluate and concentrate channels would result in a product water recovery of 50%. This flowsheet simulates a simplest setup of a one-stack ED system without any fluid recirculation, i.e., the ED stack being operated in a in-and-out single direction flow mode. This flowsheet also does not take account of the frictional pressure drop in the channel so no pump is included.  A more complicated ED flowsheet is presented as 

> 

Reviewer 2: ```suggestion
Equal flow conditions through the diluate and concentrate channels would result in a product water recovery of 50%. This flowsheet simulates the simplest setup of a one-stack ED system without any fluid recirculation, i.e., the ED stack being operated in an in-and-out single direction flow mode. This flowsheet also does not take account of the frictional pressure drop in the channel so no pump is included.  A more complicated ED flowsheet is presented as 

```

    * `One-stack electrodialysis with concentrate stream recirculation (feed-and-bleed mode) <https://watertap--1385.org.readthedocs.build/en/1385/technical_reference/flowsheets/electrodialysis_1stack_conc_recirc.html>`_.

Implementation
--------------

The modeled one-stack ED system is illustrated by Figure 1. The feed solution is split into two fluids through a separator unit, entering the diluate and concentrate channels of the ED stack. On the outlet side of the ED stack, all diluate fluids are collected into the total product water, and the concentrate fluids into the total brine stream. Water recovery in this model is volume-based, i.e., the ratio of product volume to total volume of feed solution. The model simulates the steady state of the ED system.

> 

Reviewer 2: Write the assumptions out in bullet form to be consistent with other flowsheet documentation?

.. figure:: ../../_static/flowsheets/edfs.png
    :width: 500
    :align: center

>     Figure 1. Flowsheet diagram: one-stack ED system

Reviewer 2: This figure has a mixer in the legend, but there's no mixer in the image. It seems like you may have just forgotten to delete this when creating this image from the full ED with recirculation image

> Documentation for unit models from WaterTAP: 

Reviewer 2: Add a snippet here about what configuration options should be set on the ED unit. If the defaults are used, you could say something like "The ED unit model was set up with the default configuration arguments". Otherwise, you may need to provide a bit more context as to why the configuration options in the flowsheet were selected.

    * `Electrodialysis_1D <https://watertap.readthedocs.io/en/latest/technical_reference/unit_models/electrodialysis_1D.html>`_
Documentation for unit models from IDAES: 
    * `Feed block <https://idaes-pse.readthedocs.io/en/latest/reference_guides/model_libraries/generic/unit_models/feed.html>`_
    * `Separator <https://idaes-pse.readthedocs.io/en/latest/reference_guides/model_libraries/generic/unit_models/separator.html>`_

>     * `Product block <https://idaes-pse.readthedocs.io/en/latest/reference_guides/model_libraries/generic/unit_models/product.html>`_

Reviewer 2: In other flowsheet documentation, we have not listed these IDAES models. Personally I think they can be left out since these will be present in almost every flowsheet, but I'm curious what others think.

Documentation for the property model:
    * `Multi-Component Aqueous Solution (MCAS) Property Package <https://watertap.readthedocs.io/en/latest/technical_reference/property_models/mc_aq_sol.html>`_

Degrees of Freedom
------------------

> The number of degree of freedom (DOF) is associated with the number of fixed variables (parameters) determined by the purpose of the modeling case. We implemented two modeling cases in the `main()` function: (1) the prediction of desalination outcome (salinity of the product water and saline disposal) and (2) the optimization of key decision variables in system design.  In the first case, DOF is set to zero by fixing all initial conditions of the feed solution fluid and definite ED stack parameters. All fixed values are presented in the section to follow.  In the second case, the values of those chosen to be the decision variables in the optimization are unfixed. The DOF number is therefore the number of decision variables. In this example, the decision variables are

Reviewer 2: ```suggestion
The number of degrees of freedom (DOF) is associated with the number of fixed variables (parameters) determined by the purpose of the modeling case. We implemented two modeling cases in this flowsheet: (1) the prediction of desalination outcome (salinity of the product water and saline disposal) and (2) the optimization of key decision variables in system design.  In the first case, DOF is set to zero by fixing all initial conditions of the feed solution fluid and definite ED stack parameters. All fixed values are presented in the section to follow.  In the second case, the values of those chosen to be the decision variables in the optimization are unfixed. The DOF number is therefore the number of decision variables. In this example, the decision variables are
```
It might be worth avoiding the use of `main()` since somebody reading the docs may not be familiar with what main() entails and I'm not sure if we should expect the reader to have the flowsheet code pulled up.

    * stack voltage applied
    * ED cell pair number 

> In the given optimization case, the objective function is set to LCOW and the product water salinity is set to 1 :math:`g L^{-1}` (from a feed salinity of 9.9 :math:`g L^{-1}`).

Reviewer 2: I would just copy and paste the snippet on LCOW from the BSM2 documentation, and then add what you mentioned here about the salinity

Flowsheet Specifications
------------------------

> .. csv-table:: Initial values of parameters for a 0-DOF model

Reviewer 2: I don't think this table title is necessary - the section title (Flowsheet Specifications) should suffice.
